src/arena.c

```diff
@@ -145,8 +145,18 @@ arena_stats_merge(tsdn_t *tsdn, arena_t *arena, unsigned *nthreads,
 		assert(nmalloc - ndalloc <= SIZE_T_MAX);
 		size_t curlextents = (size_t)(nmalloc - ndalloc);
 		lstats[i].curlextents += curlextents;
+
+#ifdef LIMIT_USIZE_GAP
```

Comment:
With the struct field not ifdef guarded, this can switch to a `if (config_...)` to avoid the excessive ifdefs in this file.